docs/aws/audit/kmsmonitoring/rules/secret_encrypted_kms_cmk_remediation.mdx

```diff
@@ -1,6 +1,279 @@
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a KMS Key:**
+   ```python
+   import boto3
+
+   kms_client = boto3.client('kms')
+   response = kms_client.create_key(
+       Description='My CMK for Secret Manager',
+       KeyUsage='ENCRYPT_DECRYPT',
+       Origin='AWS_KMS'
+   )
+   cmk_id = response['KeyMetadata']['KeyId']
+   print(f"Created CMK with ID: {cmk_id}")
+   ```
+
+3. **Create a Secret with CMK:**
+   ```python
+   import boto3
+
+   secrets_client = boto3.client('secretsmanager')
+   secret_name = "my_secret"
+   secret_value = "my_secret_value"
+
+   response = secrets_client.create_secret(
+       Name=secret_name,
+       SecretString=secret_value,
+       KmsKeyId=cmk_id
+   )
+   print(f"Created secret with ARN: {response['ARN']}")
+   ```
+
+### Azure (Microsoft Azure)
+
+1. **Install Azure SDK for Python:**
+   ```bash
+   pip install azure-identity azure-keyvault-secrets azure-keyvault-keys
+   ```
+
+2. **Create a Key Vault and a Key:**
+   ```python
+   from azure.identity import DefaultAzureCredential
+   from azure.keyvault.keys import KeyClient
+   from azure.keyvault.secrets import SecretClient
+
+   credential = DefaultAzureCredential()
+   key_vault_name = "myKeyVault"
+   key_vault_uri = f"https://{key_vault_name}.vault.azure.net"
+
+   key_client = KeyClient(vault_url=key_vault_uri, credential=credential)
+   key_name = "myCMK"
+   key = key_client.create_key(key_name, "RSA")
+   print(f"Created key with name: {key.name}")
+   ```
+
+3. **Create a Secret with CMK:**
+   ```python
+   secret_client = SecretClient(vault_url=key_vault_uri, credential=credential)
+   secret_name = "mySecret"
+   secret_value = "mySecretValue"
+
+   secret = secret_client.set_secret(secret_name, secret_value, key_id=key.id)
+   print(f"Created secret with name: {secret.name}")
+   ```
+
+### GCP (Google Cloud Platform)
+
+1. **Install Google Cloud SDK for Python:**
+   ```bash
+   pip install google-cloud-secret-manager google-cloud-kms
+   ```
+
+2. **Create a KMS Key:**
+   ```python
+   from google.cloud import kms_v1
+
+   client = kms_v1.KeyManagementServiceClient()
+   project_id = "my-project"
+   location_id = "global"
+   key_ring_id = "my-key-ring"
+   key_id = "my-key"
+
+   key_ring_name = client.key_ring_path(project_id, location_id, key_ring_id)
+   key_ring = client.create_key_ring(request={"parent": key_ring_name, "key_ring_id": key_ring_id})
+
+   key_name = client.crypto_key_path(project_id, location_id, key_ring_id, key_id)
+   key = client.create_crypto_key(request={"parent": key_ring_name, "crypto_key_id": key_id, "crypto_key": {"purpose": kms_v1.CryptoKey.CryptoKeyPurpose.ENCRYPT_DECRYPT}})
+   print(f"Created key with name: {key.name}")
+   ```
+
+3. **Create a Secret with CMK:**
+   ```python
+   from google.cloud import secretmanager
+
+   client = secretmanager.SecretManagerServiceClient()
+   project_id = "my-project"
+   secret_id = "my-secret"
+   secret_value = "my-secret-value"
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>